File: src/main/java/recipestore/TripleStoreEntryPoint.java
package recipestore;

import com.google.inject.Guice;
import com.google.inject.Injector;
import recipestore.input.InputModule;
import recipestore.input.RecipeApi;

class TripleStoreEntryPoint {

    public static Injector inputModule = Guice.createInjector(new InputModule());

    public static void main(String[] args) {
        loadRecipeData();
    }

    public static void loadRecipeData() {
        inputModule.getInstance(RecipeApi.class).loadRecipe(true);
    }


}

File: src/main/scala/recipestore/TripleStoreEntryPoint.scala
package recipestore

import com.google.inject.{Guice, Injector}
import recipestore.input.{InputModule, RecipeApi}

object TripleStoreEntryPoint {
  var inputModule: Injector = Guice.createInjector(new InputModule)

  def main(args: Array[String]) {
    loadRecipeData()
  }

  def loadRecipeData() {
    inputModule.getInstance(classOf[RecipeApi]).loadRecipe(true)
  }
}